File: Basics/Strings/appendChar.java
import java.io.*;
import java.util.*;

public class appendChar {

    public static Scanner scn = new Scanner(System.in);

	public static ArrayList<String> appendCharInString(String str, char ch){
		ArrayList<String> ans = new ArrayList<>();
		for(int i = 0; i <= str.length(); i++){
		    String s = str.substring(0,i) + ch + str.substring(i);
            ans.add(s);
		}
        return ans;
	}

	public static void main(String[] args) {		
		System.out.println(appendCharInString("abcde",'f'));
	}

}

File: Basics/Strings/appendChar.cpp
#include<iostream>
#include<vector>
using namespace std;
vector<string> appendCharInString(string str, char ch){
	vector<string> ans;
	for(int i = 0; i <= str.length(); i++){
	    string s = str.substr(0,i-0+1) + ch + str.substr(i);
            ans.push_back(s);
	}
    return ans;
}

int main() {		
	cout<<appendCharInString("abcde",'f');
}